tools/bin/mbedtls-rewrite-branch-style

```diff
@@ -596,6 +632,32 @@ class CodeStyleBranchRewriter(MbedTLSBranchRewriter):
         for new_commit in new_commits:
             self.restyle_commit_onto_current(new_commit)
 
+    def check_result(self, expected_content: str) -> None:
+        """Check that the resulting branch is similar to expected_content.
+
+        Similarity means that the commit history is identical except for
+        committed metadata, as with `git range-diff`.
+        """
+        assert self.target_branch is not None
+        common_ancestor = self.target_branch + '^{/Switch to the new code style}~'
+        common_ancestor_sha = self.run_git(['rev-parse', common_ancestor]).strip()
+        assert self.rewritten_sha is not None
+        self.maybe_fetch(expected_content)
+        if not self.check_git(['merge-base', '--is-ancestor',
+                               common_ancestor, expected_content]):
+            self.ok = False
+            self.log_error('{} ({}) is not an ancestor of {}',
+                           common_ancestor, common_ancestor_sha,
+                           expected_content)
+            return
+        range_diff = self.get_git_lines(['range-diff', '--no-color',
+                                         common_ancestor, expected_content, self.rewritten_sha])
+        for num, line in enumerate(range_diff, 1):
+            if not re.match(r'\A *[0-9]+: *[0-9a-f]+ *= ', line):
+                self.ok = False
+                self.log_error('Bad range-diff line {}: {}', num, line)
+        self.info('Check passed: {} === {}', self.rewritten_sha, expected_content)
```

Comment:
Shouldn't this be printed only if self.ok is not False? Or renamed to "Check results:"?